dotCMS/src/main/java/com/dotmarketing/business/UserAPIImpl.java

```diff
@@ -186,106 +201,119 @@ public User getAnonymousUser() throws DotDataException {
     }
     
 
-
+	@WrapInTransaction
 	private User _getAnonymousUser() throws DotDataException {
 		User user = null;
 		try {
-			user = uf.loadUserById("anonymous");
+			user = userFactory.loadUserById("anonymous");
 		} catch (DotDataException e) {
 			user = createUser("anonymous", "anonymous@dotcmsfakeemail.org");
 			user.setUserId("anonymous");
 			user.setFirstName("anonymous user");
 			user.setCreateDate(new java.util.Date());
 			user.setCompanyId(PublicCompanyFactory.getDefaultCompanyId());
-			uf.saveUser(user);
+			userFactory.saveUser(user);
 			com.dotmarketing.business.APILocator.getRoleAPI().addRoleToUser(com.dotmarketing.business.APILocator.getRoleAPI().loadRoleByKey(Config.getStringProperty("CMS_ANONYMOUS_ROLE")).getId(), user);
 		} catch (NoSuchUserException e) {
 			user = createUser("anonymous", "anonymous@dotcmsfakeemail.org");
 			user.setUserId("anonymous");
 			user.setFirstName("anonymous user");
 			user.setCreateDate(new java.util.Date());
 			user.setCompanyId(PublicCompanyFactory.getDefaultCompanyId());

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "PublicCompanyFactory"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)